gremlin-go/driver/graphTraversal.go

```diff
@@ -670,3 +675,99 @@ func (g *GraphTraversal) Write(args ...interface{}) *GraphTraversal {
 	g.bytecode.addStep("write", args...)
 	return g
 }
+
+type transaction struct {
+	g                      *GraphTraversalSource
+	sessionBasedConnection *DriverRemoteConnection
+	remoteConnection       *DriverRemoteConnection
+	isOpen                 bool
+	mutex                  sync.Mutex
+}
+
+func (t *transaction) Begin() (*GraphTraversalSource, error) {
```

Comment:
Double check the exporting of these methods. I think they don't need to be.

Reply:
They do need to be